apps/storefront/src/utils/headlessInitializer.ts

```diff
@@ -0,0 +1,79 @@
+import { getAPIBaseURL } from '../shared/service/request/base';
+import { Environment } from '../types';
+
+const BUYER_PORTAL_INJECTED_SCRIPT_CLASS = `buyer-portal-scripts-headless`;
+
+export async function initHeadlessScripts() {
+  const parseAndInsertStorefrontScripts = (storefrontScripts: string) => {
+    const b2bScriptDocument = new DOMParser().parseFromString(storefrontScripts, 'text/html');
+    const b2bScriptNodes = b2bScriptDocument.querySelectorAll('script');
+
+    if (b2bScriptNodes.length > 0) {
+      const existingBuyerPortalScriptNodes = document.querySelectorAll(
+        `script.${BUYER_PORTAL_INJECTED_SCRIPT_CLASS}`,
+      );
+      if (existingBuyerPortalScriptNodes.length > 0) {
+        existingBuyerPortalScriptNodes.forEach((oldNode) => {
+          oldNode.parentNode?.removeChild(oldNode);
+        });
+      }
+
+      b2bScriptNodes.forEach((scriptNode) => {
+        const newScriptElement = document.importNode(scriptNode, true);
+        newScriptElement.className = BUYER_PORTAL_INJECTED_SCRIPT_CLASS;
+
+        document.body.appendChild(newScriptElement);
```

Comment:
biggest change is using `document.importNode` https://developer.mozilla.org/en-US/docs/Web/API/Document/importNode

so that we don't manually have to parse different attributes.